### PR TITLE
[CALCITE-2247] Simplify AND and OR conditions using predicates.

### PR DESCRIPTION
**Pull request by kgyrtkirk**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rex/RexSimplify.java [hunk 1]
```diff
@@ -312,13 +361,38 @@ private RexNode simplifyNot(RexCall call) {
...
+
+    for (RexNode p : predicates.pulledUpPredicates) {
+      IsPredicate pred = IsPredicate.of(p);
+      if (pred == null || !a.toString().equals(pred.ref.toString())) {
+        continue;
+      }
+      if (kind == pred.kind) {
+        return rexBuilder.makeLiteral(true);
+      } else {
+        return rexBuilder.makeLiteral(false);
```
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/rex/RexSimplify.java]: This is wrong.

Counter-example is `OR(IS NOT NULL($10), IS NOT FALSE($10))` where `$10` is a nullable boolean.

It is trying to simplify `is not false($10)` with assumption of `NOT(is not null($10))` which is `is null($10)`.
It sees that it has assumption for `IS NULL($10)` and its kind does not match to `IS NOT FALSE`, thus it simplifies the predicate to FALSE which is wrong.
`IS NOT FALSE(null)` yields `true`.

**Comment by kgyrtkirk** [comment]
Yes, the condition tried to identify it is the same kind of 'is' or not...unfortunately I  wasn't taking into account 'is false'

File: core/src/main/java/org/apache/calcite/rex/RexSimplify.java (hunk 1 above)
Comment (jcamachor) [reply core/src/main/java/org/apache/calcite/rex/RexSimplify.java]: @vlsi , was this issue ever addressed? Going through this logic to fix CALCITE-2639 .

Comment (vlsi) [reply core/src/main/java/org/apache/calcite/rex/RexSimplify.java]: I've no idea.

**Pull request closed without merging**